fava_investor/modules/performance/balances.py

```diff
@@ -0,0 +1,65 @@
+import re
+
+from beancount.core.account import parent, parents
+from fava.core import FavaLedger, Tree
+from fava.core.tree import TreeNode
+from fava_investor.common.favainvestorapi import FavaInvestorAPI
+
+
+def get_closed_tree_with_value_accounts_only(accapi: FavaInvestorAPI, config) -> Tree:
+    ledger: FavaLedger = accapi.ledger
+    tree = ledger.root_tree_closed
+    accounts_to_keep = get_value_accounts_and_parents(accapi.ledger.accounts, config.get('value', ['.*']))
+    filter_tree(tree, accounts_to_keep)
+    return tree
+
+
+def remove_account_from_tree(tree: Tree, account: str):
+    if account not in tree or account == '':
+        return
+    node = tree[account]
+    for child in list(node.children):
```

Comment:
Is removing children needed? Won't it suffice to simply remove this node from its parent?

And perhaps in addition simply do a `del tree[account]` for each child account, if that is needed?

Reply:
its also completely removing children of children (...) from the tree as well. If I didn't remove them they wouldn't be rendered in template (no parent accounts pointing to them) but I thought it would be messy to leave them partially in the tree.